### ISSUE TITLE
[Windows] Could not add Windows worker with Azure Cloud Provider

### ISSUE THREAD

**Issue by thxCode** [rancher/rancher#22406]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Setup rancher/rancher:master-head
2. Bring up a control plane + etcd + Linux worker with Azure cloud provider
3. Add a Windows worker

**Result:**
Could not add Windows worker successfully

**Other details that may be helpful:**
1. `docker logs kubelet --tail 50`:
```
W0824 13:26:49.527224   10680 plugins.go:118] WARNING: azure built-in cloud provider is now deprecated. The Azure provider is deprecated and will be removed in a future release
F0824 13:26:49.528226   10680 server.go:273] failed to run Kubelet: could not init cloud provider "azure": autorest/azure: There is no cloud environment matching the name "AZURECLOUD"
```
2. `cat C:\etc\kubernetes\cloud-config`:
```
{...,"cloud":"AzureCloud",...}
```

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): rancher/rancher:master-head
- Installation option (single install/HA): single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud
- Kubernetes version (use `kubectl version`): v1.15.3

**Comment by izaac** [comment]
Rancher version: master-head (v2.3) (08/28/2019)

I was able to add a windows worker after the change in Azure hosted Windows Server 1903  with docker 19.03.1

Rancher displayed it as Active.
Linux workers also were Active

<img width="1261" alt="Screen Shot 2019-08-28 at 5 40 22 PM" src="https://user-images.githubusercontent.com/27659/63901838-34001a00-c9bb-11e9-9d3c-b2d524efadb5.png">

@sangeethah @thxCode

**Comment by davidnuzik** [comment]
As we discussed offline, please ensure that you have cloud option set to Azure and the windows worker still comes up as Active. Ensure you can leverage an azure cloud feature (to ensure it's working correctly) such as an Azure Disk via storage class.

Ensure you don't see the second line in kubelet logs that is shown in the issue description.

**Comment by izaac** [comment]
@davidnuzik I tested with the original value of `None` as we originally identified the issue without setting the cloud provider.

That test passed.

As suggested I used `Azure` as the cloud provider today and added a PVC and double checked with @sowmyav27 that the volume was correctly created and shared between pods in Windows nodes.

That also passed.

Looking at the `kubelet` logs I no longer see the errors reported before the fix.

Rancher version: master-head (v2.3) (08/29/2019) 
Docker 19.03.1
Windows server core 1903.

@sangeethah I'm closing this issue now as we covered the initial scope and the suggested extended coverage to validate Azure cloud functionality is working properly.
